src/foam/nanos/auth/ruler/UserLifecycleTicketSink.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright 2024 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth.ruler',
+  name: 'UserLifecycleTicketSink',
+  extends: 'foam.dao.AbstractSink',
+
+  documentation: 'Desgined to be used by the UserLifecycleTicket to delete or disable user associated models',
+
+  javaImports: [
+    'foam.core.FObject',
+    'foam.core.PropertyInfo',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.dao.ProxyDAO',
+    'foam.dao.Sink',
+    'foam.nanos.auth.EnabledAware',
+    'foam.nanos.auth.LifecycleAware',
+    'foam.nanos.auth.LifecycleState',
+    'foam.nanos.auth.User',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.logger.Loggers',
+    'java.util.List'
+  ],
+
+  javaCode: `
+    public UserLifecycleTicketSink(X x, String daoKey) {
+      setX(x);
+      setDaoKey(daoKey);
+    }
+
+    public UserLifecycleTicketSink(X x, LifecycleState state, String daoKey) {
+      setX(x);
+      setLifecycleState(state);
+      setDaoKey(daoKey);
+    }
+  `,
+
+  properties: [
+    {
+      class: 'String',
+      name: 'daoKey'
+    },
+    {
+      class: 'Enum',
+      of: 'foam.nanos.auth.LifecycleState',
+      name: 'lifecycleState',
+      value: 'DELETED'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'put',
+      javaCode: `
+      X x = getX();
+      Logger logger = (Logger) x.get("logger");
+      PropertyInfo prop = (PropertyInfo) ((FObject)obj).getClassInfo().getAxiomByName("id");
+      Object id = prop != null ? prop.get(obj) : null;
```

Comment:
FObjects that has "id" property are implementing foam.core.Identifiable interface. We may get the primary from it directly.
```suggestion
      Object id = obj instanceof Identifiable ? ((Identifiable) obj).getPrimaryKey() : null;
```